src/backend/expungeservice/expunger/analyze.py

```diff
@@ -118,3 +118,182 @@ def __init__(self, name, dob, cases):
+        check = ('Does the charge fall under any of these statutes: ' +
+                 ','.join(str(statute) for statute in statutes))
+        # TODO implement this
+        return Check(check=check, result=False, check_desc=desc)
+
+    def _is_charge_sex_crime(charge):
+        # TODO update
+        _statutes_sex_crimes = []
+        return RecordAnalyzer._is_charge_in_statute_list(
+            charge, _statutes_sex_crimes, 'Is the charge a sex crime')
+
+    def _is_charge_traffic_crime(charge):
+        # TODO update
+        _statutes_traffic_crimes = []
+        return RecordAnalyzer._is_charge_in_statute_list(
+            charge, _statutes_traffic_crimes, 'Is the charge a traffic crime')
+
+    def _have_open_case(self):
+        check = 'Is there a open case for the client'
+        result = any([case.state == CaseState.OPEN for case in self.client.cases])
+        return Check(check=check, result=result)
+
+    """
+    Run Time Eligibility analysis on the client and their records
+
+    TODO: make it return analysis for each charge as well (which is supposed
+    to be an update on eligibility date and relevan statutes)
+
+    Returns:
+        An Result instance
+    """
+    def time_eligibility(self):
+        checks = []
+
+        checks.append(self._have_open_case())
+        if checks[-1].result:
+            return Result(ResultCode.OPEN_CASE, checks)
+
+        # TODO implement the rest
+
+        return Result(ResultCode.NO_ACTION)
+
+    """
+    Run Type Eligibility analysis on a charge
+
+    Args:
+        charge: A Charge instance
+
+    Returns:
+        A Result instance
+    """
+    def type_eligibility(self, charge):
+        checks = []
+
+        checks.append(RecordAnalyzer._is_charge_level(charge, 'Felony', 'A'))
+        if checks[-1].result:
+            return ResultInElig137(checks=checks)
+
+        checks.append(RecordAnalyzer._is_charge_sex_crime(charge))
+        if checks[-1].result:
+            return ResultInElig137(checks=checks)
+
+        checks.append(RecordAnalyzer._is_charge_traffic_crime(charge))
+        if checks[-1].result:
+            return ResultInElig137(checks=checks)
+
+        # TODO add remaining checks
+
+        return Result(ResultCode.FURTHER_ANALYSIS, checks)
+
+    """
+    Analyze which records are expungeable
+
+    Returns:
+        A tuple consisting of:
+            - A list of ChargeAnalyis instances that describes Type Eligibility
+              analysis for each charge.
+            - An Result instance that describes the Time Eligibility analyis.
+    """
+    def analyze(self):
+        type_elig_analysis_list = []
+        for case in self.client.cases:
+            for charge in case.charges:
+                result = self.type_eligibility(charge)
+                type_elig_analysis_list.append(ChargeAnalysis(charge, case, result))
+
+        time_elig_analysis = self.time_eligibility()
+        return type_elig_analysis_list, time_elig_analysis
```

Comment:
It may be better to add the expunge-ability of a charge onto the charge class itself as an attribute. This will help with presenting the information to the user. And then when we test we can do something like `charge.result.code`

Reply:
I thought about that and went back and forth between modifying the charge object itself vs returning a new object (ChargeAnalysis) that has the result and reference to that charge. It seemed the latter is cleaner as it doesn't have side effects. What do you think?

Reply:
I see. I'm not against side effects as some side effects are good. I think the UI will ultimately decide on how we proceed here and if this data ever plans to go in a database will also be a factor.